token/program-2022-test/tests/scaled_ui_amount.rs

```diff
@@ -0,0 +1,343 @@
+#![cfg(feature = "test-sbf")]
+
+mod program_test;
+use {
+    program_test::{keypair_clone, TestContext, TokenContext},
+    solana_program_test::{
+        processor,
+        tokio::{self, sync::Mutex},
+        ProgramTest,
+    },
+    solana_sdk::{
+        account_info::{next_account_info, AccountInfo},
+        entrypoint::ProgramResult,
+        instruction::{AccountMeta, Instruction, InstructionError},
+        msg,
+        program::{get_return_data, invoke},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+        signature::Signer,
+        signer::keypair::Keypair,
+        transaction::{Transaction, TransactionError},
+        transport::TransportError,
+    },
+    spl_token_2022::{
+        error::TokenError,
+        extension::{scaled_ui_amount::ScaledUiAmountConfig, BaseStateWithExtensions},
+        instruction::{amount_to_ui_amount, ui_amount_to_amount, AuthorityType},
+        processor::Processor,
+    },
+    spl_token_client::token::{ExtensionInitializationParams, TokenError as TokenClientError},
+    std::{convert::TryInto, sync::Arc},
+};
+
+#[tokio::test]
+async fn success_initialize() {
```

Comment:
Just for completeness, should we add a test to invoke the newly added error variant `TokenError::InvalidScale` by trying to initialize with an invalid scale like 0?

Reply:
ah yes, good idea, I forgot to add that, thanks!